.openapidoc/index.js

```diff
@@ -0,0 +1,167 @@
+const fs = require("fs");
+const yaml = require("js-yaml");
+const { https } = require("follow-redirects");
+const ghpages = require("gh-pages");
+
+// constants
+const repo = "git@github.com:PegaSysEng/eth2signer.git";
+const branch = "gh-pages";
+const spec = "../core/build/resources/main/openapi/eth2signer.yaml";
+const versionsFileUrl =
+  "https://github.com/PegaSysEng/eth2signer/raw/gh-pages/versions.json";
+const distDir = "./dist";
+const versionsFile = distDir + "/versions.json";
+
+/**
+ * Downloads the given source URL and return it as a string
+ * @param source
+ */
+async function downloadAsString(source) {
+  return new Promise((resolve, reject) => {
```

Comment:
I'd just use `fetch` as it returns a promise, so you can `await` it.

Reply:
Here's an example of how I use `fetch` in the `circle-build-monitor` - it's TypeScript, but TS is just a superset of JavaScript, so it should be easy enough to port over... https://github.com/PegaSysEng/circle-build-monitor/blob/master/src/polling.ts#L122

Reply:
I've switched to `node-fetch` npm module instead, this method is removed.